.teamcity/_self/PluginBaseBuild.kt

```diff
@@ -53,12 +54,26 @@ open class PluginBaseBuild : Template({
 				}
 			}
 		}
+
+		sshAgent {
+			id = "WpCalypsoGitSSH"
+			teamcitySshKey = "Sergio TeamCity"
+		}
 	}
 
 	steps {
 		bashNodeScript {
 			name = "Prepare environment"
 			scriptContent = """
+				# Merge the target branch first.
+				if [[ "%teamcity.build.branch.is_default%" != "true" ]] ; then
+					ssh-agent
+					ssh-add -l
+					git fetch origin trunk
```

Comment:
Do we need this if we also have `teamcity.git.fetchAllHeads`? I'd assume that at least `trunk` is always being fetched.

Btw, technically we should merge in the _target_ branch of the PR, which may not be trunk. But that could be lots of complexity (I guess we'll need to use GH API to find out the target branch) and it will just using trunk 99.99% of the cases.

Reply:
Yeah, that's what I was thinking too. If the TC builds were triggered by PRs, then we would have the target branch available as an environment parameter, but we trigger builds based on commits pushed to branches, so I'm thinking that param won't exist.

I also wonder if we want to use trunk anyways, since the goal of the notifications is to detect if the current build is different compared to trunk. So we want to make sure the "latest trunk release changes" are incorporated into our branch, which would always mean merging trunk instead of the target branch. 🤔

Reply:
> Do we need this if we also have teamcity.git.fetchAllHeads?

Not sure, but it would be worth a test. Maybe in that case, we don't need to worry about the SSH stuff!

Reply:
Yep, that worked perfectly. Thanks for putting me on the right track :)